src/target/at32f43x.c

```diff
@@ -129,6 +148,48 @@ static void at32f43_add_flash(target_s *const target, const target_addr_t addr,
 	target_add_flash(target, target_flash);
 }
 
+static bool at32f43_configure_dbgmcu(target_s *target)
+{
+	/* Collect target_mem errors by |= */
+	bool error = false;
+	/*
+	 * Enable sleep state emulation (clocks fed by HICK)
+	 * and make both watchdogs pause during core halts so that they
+	 * don't issue extra resets when we're doing e.g. flash reprogramming
+	 */
+	const uint32_t dbgmcu_ctrl = target_mem32_read32(target, AT32F43x_DBGMCU_CTRL);
+	if ((dbgmcu_ctrl & AT32F43x_DBGMCU_CTRL_SLEEP_Msk) == 0U)
```

Comment:
It doesn't matter if any of the bits are already set, we need to ensure all of them are so please either check that the value is not already the mask value, or elide the `if`. We prefer elision of the `if` for code size and complexity reduction.

Ditto for the pause mask check below.

With the if elided, we would prefer to see each of the debug bits being set individually as in `target_mem32_write32(target, AT32F43x_DBGMCU_CTRL, target_mem32_read32(target, AT32F43x_DBGMCU_CTRL) | AT32F43x_DBGMCU_CTRL_SLEEP_DEBUG | AT32F43x_DBGMCU_CTRL_DEEPSLEEP_DEBUG | AT32F43x_DBGMCU_CTRL_STANDBY_DEBUG);` so it's a little clearer what exactly is getting set here.

Reply:
Oh, I did the check correctly for APB1_PAUSE, but forgot about CTRL. Enhanced the comparison.
Note that there are only these three significant bits in CTRL register. I intended to elide the SWD writes, which could be about 50 microseconds long each (assuming two transactions, the second checking/clearing error).